jwst/resample/resample.py

```diff
@@ -275,17 +286,19 @@ def resample_variance_array(self, name, output_model):
             self.drizzle_arrays(variance, inwht, model.meta.wcs,
                                 output_wcs, resampled_variance, outwht, outcon,
                                 pixfrac=self.pixfrac, kernel=self.kernel,
-                                fillval=np.inf)
+                                fillval=np.nan)
 
             # Add the inverse of the resampled variance to a running sum
-            with np.errstate(divide="ignore"):
-                inverse_variance_sum += np.reciprocal(resampled_variance)
+            mask = (outwht > 0) & (resampled_variance > 0)
```

Comment:
To save me rummaging through other code can you say where `outwht` gets set to non-zero values? Presumably in the above `self.drizzle_arrays` call?

Reply:
I simplified mask computation. Checking for `resampled_variance > 0` should be enough.